sycl/test-e2e/Matrix/SG32/joint_matrix_bf16_fill_k_cache_unroll.cpp

```diff
@@ -10,7 +10,7 @@
 // RUN: %if gpu %{ %{build} -mllvm -inline-threshold=5000 -o %t_gpu.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4 -DMANUAL_UNROLL %}
 // RUN: %if gpu %{ %{run} %t_gpu.out %}
 
-// RUN: %if cpu %{ %{build} -mllvm -inline-threshold=5000 -o %t_cpu.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4 -DMANUAL_UNROLL -DtM=16 -DtK=32 -DNCACHE1=32 -DKCACHE1=32
+// RUN: %if cpu %{ %{build} -mllvm -inline-threshold=5000 -o %t_cpu.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4 -DMANUAL_UNROLL -DtM=16 -DtK=32 -DNCACHE1=32 -DKCACHE1=32 %}
 // RUN: %if cpu %{ %{run} %t_cpu.out %}
 
 // XFAIL: gpu
```

Comment:
// RUN: %{build} -mllvm -inline-threshold=5000 -o %t_gpu.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4 -DMANUAL_UNROLL
// RUN: %{build} -mllvm -inline-threshold=5000 -o %t_cpu.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=4 -DMANUAL_UNROLL -DtM=16 -DtK=32 -DNCACHE1=32 -DKCACHE1=32

// RUN: %if gpu %{ %{run} %t_gpu.out %}
// RUN: %if cpu %{ %{run} %t_cpu.out %}

Reply:
Hi @dkhaldi , I checked that adding %} at the end is not enough to fix these tests.
if gpu/if cpu doesn't work for builds, so we should build for gpu and cpu unconditionally.
Then run either cpu or gpu, when condition satisfied.
Also, please, remote xfail:gpu.
See the code I inserted above

Reply:
The same should be applied for all 4 tests.